pkg/kn/flags/podspec.go

```diff
@@ -163,3 +170,165 @@ func (p *PodSpecFlags) AddFlags(flagset *pflag.FlagSet) []string {
+			}
+		}
+
+		err := UpdateEnvFrom(podSpec, envFromSourceToUpdate, envFromSourceToRemove)
+		if err != nil {
+			return err
+		}
+	}
+
+	if cmd.Flags().Changed("mount") || cmd.Flags().Changed("volume") {
+		mountsToUpdate, mountsToRemove, err := util.OrderedMapAndRemovalListFromArray(p.Mount, "=")
+		if err != nil {
+			return fmt.Errorf("Invalid --mount: %w", err)
+		}
+
+		volumesToUpdate, volumesToRemove, err := util.OrderedMapAndRemovalListFromArray(p.Volume, "=")
+		if err != nil {
+			return fmt.Errorf("Invalid --volume: %w", err)
+		}
+
+		err = UpdateVolumeMountsAndVolumes(podSpec, mountsToUpdate, mountsToRemove, volumesToUpdate, volumesToRemove)
+		if err != nil {
+			return err
+		}
+	}
+
+	if cmd.Flags().Changed("image") {
+		err = UpdateImage(podSpec, p.Image.String())
+		if err != nil {
+			return err
+		}
+	}
+
+	if cmd.Flags().Changed("limits-cpu") || cmd.Flags().Changed("limits-memory") {
+		if cmd.Flags().Changed("limit") {
+			return fmt.Errorf("only one of (DEPRECATED) --limits-cpu / --limits-memory and --limit can be specified")
+		}
+		fmt.Fprintf(cmd.OutOrStdout(), "\nWARNING: flags --limits-cpu / --limits-memory are deprecated and going to be removed in future release, please use --limit instead.\n\n")
+	}
+
+	if cmd.Flags().Changed("requests-cpu") || cmd.Flags().Changed("requests-memory") {
+		if cmd.Flags().Changed("request") {
+			return fmt.Errorf("only one of (DEPRECATED) --requests-cpu / --requests-memory and --request can be specified")
+		}
+		fmt.Fprintf(cmd.OutOrStdout(), "\nWARNING: flags --requests-cpu / --requests-memory are deprecated and going to be removed in future release, please use --request instead.\n\n")
+	}
+
+	limitsResources, err := p.computeResources(p.LimitsFlags)
+	if err != nil {
+		return err
+	}
+	requestsResources, err := p.computeResources(p.RequestsFlags)
+	if err != nil {
+		return err
+	}
+	err = UpdateResourcesDeprecated(podSpec, requestsResources, limitsResources)
+	if err != nil {
+		return err
+	}
+
+	requestsToRemove, limitsToRemove, err := p.Resources.Validate()
+	if err != nil {
+		return err
+	}
+
+	err = UpdateResources(podSpec, p.Resources.ResourceRequirements, requestsToRemove, limitsToRemove)
+	if err != nil {
+		return err
+	}
+
+	if cmd.Flags().Changed("cmd") {
+		err = UpdateContainerCommand(podSpec, p.Command)
+		if err != nil {
+			return err
+		}
+	}
+
+	if cmd.Flags().Changed("arg") {
+		err = UpdateContainerArg(podSpec, p.Arg)
+		if err != nil {
+			return err
+		}
+	}
+
+	if cmd.Flags().Changed("port") {
+		err = UpdateContainerPort(podSpec, p.Port)
+		if err != nil {
+			return err
+		}
+	}
+
+	if cmd.Flags().Changed("service-account") {
+		err = UpdateServiceAccountName(podSpec, p.ServiceAccountName)
+		if err != nil {
+			return err
+		}
+	}
+
+	if cmd.Flags().Changed("pull-secret") {
+		UpdateImagePullSecrets(podSpec, p.ImagePullSecrets)
+	}
+
+	if cmd.Flags().Changed("user") {
+		UpdateUser(podSpec, p.User)
```

Comment:
It'd nice to handle error from `UpdateUser()` as well to be on pair with other methods.

Reply:
thank you, fixed.